File: app/src/main/java/closer/vlllage/com/closer/api/models/GroupMemberResult.java
package closer.vlllage.com.closer.api.models;

import closer.vlllage.com.closer.store.models.GroupMember;

public class GroupMemberResult extends ModelResult {
    public String from;
    public String to;
    public boolean muted;
    public boolean subscribed;

    public static GroupMember from(GroupMemberResult groupMemberResult) {
        GroupMember groupMember = new GroupMember();
        groupMember.setId(groupMemberResult.id);
        updateFrom(groupMember, groupMemberResult);
        return groupMember;
    }

    public static GroupMember updateFrom(GroupMember groupMember, GroupMemberResult groupMemberResult) {
        groupMember.setGroup(groupMemberResult.to);
        groupMember.setPhone(groupMemberResult.from);
        groupMember.setMuted(groupMemberResult.muted);
        groupMember.setSubscribed(groupMemberResult.subscribed);
        return groupMember;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/api/models/GroupMemberResult.kt
package closer.vlllage.com.closer.api.models

import closer.vlllage.com.closer.store.models.GroupMember

class GroupMemberResult : ModelResult() {
    var from: String? = null
    var to: String? = null
    var muted: Boolean = false
    var subscribed: Boolean = false

    companion object {

        fun from(groupMemberResult: GroupMemberResult): GroupMember {
            val groupMember = GroupMember()
            groupMember.id = groupMemberResult.id
            updateFrom(groupMember, groupMemberResult)
            return groupMember
        }

        fun updateFrom(groupMember: GroupMember, groupMemberResult: GroupMemberResult): GroupMember {
            groupMember.group = groupMemberResult.to
            groupMember.phone = groupMemberResult.from
            groupMember.muted = groupMemberResult.muted
            groupMember.subscribed = groupMemberResult.subscribed
            return groupMember
        }
    }
}
